examples/with-segment-analytics/components/analytics.tsx

```diff
@@ -0,0 +1,16 @@
+'use client'
+
+import { useEffect } from 'react'
+import { usePathname, useSearchParams } from 'next/navigation'
+import { analytics } from '@/lib/segment'
+
+export default function Analytics() {
+  const pathname = usePathname()
+  const searchParams = useSearchParams()
+
+  useEffect(() => {
+    analytics.page()
+  }, [pathname, searchParams])
```

Comment:
Tiny heads up here. I was seeing double tracking calls here. You can reduce the double tracking calls by doing this:
```
import { useEffect } from 'react';
import { usePathname, useSearchParams } from 'next/navigation';
const pathname = usePathname();
const searchParams = useSearchParams();
const paramsToString = searchParams.toString();
useEffect(() => {
    analytics.page();
}, [pathname, paramsToString]);
```